### PR TITLE
cmd/build: Reduce duplicate code

### PR DESCRIPTION
**Pull request by mauriciovasquezbernal**
- Use function to generate build arguments
- Copy makefiles also when building in a container
- Move all helpers files (Makefiles) to folder and use embed.FS

/cc @blanquicet

### CODE REVIEW

File: Dockerfiles/ebpf-builder.Dockerfile [hunk 1]
```diff
@@ -71,5 +71,3 @@ WORKDIR /work
 
 # Add files used to build containerized gadgets
 ADD include /usr/include
-ADD cmd/common/image/Makefile.build /Makefile
-ADD cmd/common/image/Makefile.build.btfgen /Makefile.build.btfgen
```
[the remark this answers is not in the document]
Comment (blanquicet) [reply Dockerfiles/ebpf-builder.Dockerfile:75]: What is the use case where one could need to build the gadget in such a way?

[the remark this answers is not in the document]
Comment (mauriciovasquezbernal) [reply Dockerfiles/ebpf-builder.Dockerfile:75]: This makefile in an internal implementation detail of IG, users relying on this should bring their own makefile after this PR.

**Comment by mauriciovasquezbernal** [comment]
> Hi!
> 
> I am not sure to understand what is the final goal. Particularly, adding the `buildCmd()` is not really linked to remove the Makefiles from the image. Can you please share more insights on the goal?
> 
> Best regards.

The idea is to have the `ig image build` command self-contained without depending on Makefiles already present on the builder image. 

Using the makefiles on the eBPF builder image has the following issues:
- It introduces another dependency on the version of the image, if we change something on those makefiles, then the user will have to bump the version of the image they're using
- introducing more helper files, like done in #3372, implies adding a lot of lines around

**Review by blanquicet — APPROVED** [review]
LGTM! I tested it building with the ebpf-builder image and locally and it worked properly.

**Pull request merged**